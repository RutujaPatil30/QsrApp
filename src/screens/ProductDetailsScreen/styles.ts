import { StyleSheet } from "react-native";
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';
import theme from "../../utils/themes";


const styles = StyleSheet.create({
    container :{
        flex:1, 
        backgroundColor:theme.colors.primary,
    },
    imageContainer:{
        height:hp('30%'),
        backgroundColor:theme.colors.primary,
        alignItems:'center'
    },
    imageStyles:{
        top:hp('8%'),
        resizeMode:'contain',
        width : hp('40%'),
        height: hp('30%'),
    },
    detailsContainer:{
        flex:1,
        paddingTop:hp('9%'),
    },
    bottomContainer:{
        top: hp('24%'),
        backgroundColor: theme.colors.white,
        borderTopRightRadius: hp('5%'),
        borderTopLeftRadius: hp('5%'),
        padding:hp('2.5%'),
    },
    productTitle:{
        fontSize:theme.fonts.title,
        fontWeight:'bold',
        color:theme.colors.black
    },
    priceLabel:{
        fontSize:theme.fonts.detailsLabel,
        color:theme.colors.secondary,
        fontWeight:'bold'
    },
    productdescription:{
        fontSize:theme.fonts.subtitle,
        paddingVertical:hp('1.5%'),
    },
    optionalContainer:{
        flexDirection:'row',
        alignItems:'center',
    },
    optionalText:{
        fontSize:theme.fonts.detailsLabel,
        color: theme.colors.black,
        fontWeight:'bold'
    },
    optionalLabel:{
        fontSize:theme.fonts.subtitle,
        fontWeight:'bold',
        marginLeft:hp('0.5%'),
    },
    textInputStyle:{
        backgroundColor: theme.colors.lightGrey,
        borderRadius: hp('1.5%'),
        alignSelf: 'center',
        width: wp('90%'),
        marginVertical:hp('2.5%'),
        padding:hp('1.5%')
    },
    buttonContainer:{
        flexDirection:'row',
        padding: hp('1.5%'),
        alignItems: 'center',
        justifyContent:'space-evenly',
        backgroundColor:theme.colors.white
    },
    iconContainer:{
        backgroundColor:theme.colors.primary,
        borderRadius:wp('15%'),
        justifyContent:'center',
        alignItems:'center',
        height: hp('5.7%'),
        width: wp('13%'),
    },
    animationView: {
        position: 'absolute',
        left: 6,
        right: 0,
        alignItems: 'center',
      },
})

export default styles;