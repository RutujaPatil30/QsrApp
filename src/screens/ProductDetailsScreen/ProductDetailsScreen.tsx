import React, { useRef } from "react"
import { Text, TextInput, TouchableOpacity } from "react-native";
import { Image, View } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import * as Animatable from 'react-native-animatable';
import Icon from "react-native-vector-icons/MaterialCommunityIcons";
import CustomButton from "../../components/CustomButton/CustomButton";
import baselocalization from "../../utils/baselocalization";
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import theme from "../../utils/themes";
import Toast from 'react-native-simple-toast';
import styles from "./styles";
import { useDispatch } from "react-redux";
import { add_to_cart_request } from "../../redux/action";

const ProductDetailsScreen = (props: any) => {
    const { navigation, route } = props;
    const dispatch = useDispatch();
    const [text, setText] = React.useState('');

    const addItems = () => {
        dispatch(add_to_cart_request(route.params.itemDetails));
        Toast.show(baselocalization.productDetailsScreen.productsDetailsToast, Toast.SHORT);
    };
    
    return (
        <View style={styles.container}>
            <Animatable.View animation="fadeIn" style={{ flex: 1 }}>
                <View style={styles.detailsContainer}>
                    <ScrollView style = {styles.bottomContainer}>
                    <Text style={styles.productTitle}>{route.params.name}</Text>
                    <Text style={styles.priceLabel}>Rs. {route.params.price}</Text>
                    <Text style={styles.productdescription}>{route.params.description}</Text>
                    <View style={styles.optionalContainer}>
                        <Text style={styles.optionalText}>{baselocalization.productDetailsScreen.productDetailsOptionalText}</Text>
                        <Text style={styles.optionalLabel}>{baselocalization.productDetailsScreen.productDetailsOptionalLabel}</Text>
                    </View>
                    <TextInput
                        placeholder={baselocalization.productDetailsScreen.productsDetailsTextInput}
                        placeholderTextColor={theme.colors.grey}
                        value={text}
                        onChangeText={text => setText(text)}
                        style={styles.textInputStyle}
                    />
                    </ScrollView>
                </View>
                <View style={styles.buttonContainer}>
                    <TouchableOpacity
                        style={styles.iconContainer}
                        onPress={addItems}>
                        <Icon
                            name="plus"
                            size={25}
                            color={theme.colors.white}
                        />
                    </TouchableOpacity>
                    <CustomButton text={baselocalization.onboardingScreen.onboardingScreenButtonText}
                        disabled={false}
                        buttonStylesProps={{
                            backgroundColor: theme.colors.primary,
                            borderColor: theme.colors.primary,
                            textColor: theme.colors.white,
                            width: wp('75%'),
                            height: hp('6%'),
                            fontSize: theme.fonts.labels
                        }}
                        onPress={()=>{navigation.navigate('CartScreen')}}
                    />
                </View>
                <Animatable.View animation="slideInUp" style={styles.animationView}>
                    <Image source={route.params.image} style={styles.imageStyles} />
                </Animatable.View>
            </Animatable.View>
        </View>
    )
}

export default ProductDetailsScreen;