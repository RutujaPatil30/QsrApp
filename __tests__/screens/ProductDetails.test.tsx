
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import ProductDetailsScreen from '../../src/screens/ProductDetailsScreen/ProductDetailsScreen';
import { NavigationContainer} from '@react-navigation/native';
import { Provider } from 'react-redux';
import store from '../../src/redux/store';
import renderer from 'react-test-renderer';

describe('Checking ProductDetailsScreen ', () => {
  it('should render the UI "ProductDetailsScreen" ', () => {

    const mockNavigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };

    const mockRoute = {
      params: {
        id: '123',
        name: 'Big Mac',
        price: '5.99',
        description: 'A big, tasty burger with two beef patties and special sauce.',
        image: require('../../src/assets/breakfast.png'),
        
      },
    };

    const component = (
      <Provider store={store}>
        <NavigationContainer>
          <ProductDetailsScreen navigation={mockNavigation} route={mockRoute} />
        </NavigationContainer>
      </Provider>
    );
    render(component);

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();

    const inputField = screen.getByTestId('ProductDetailsScreen_InputField');
    const nextButton = screen.getByTestId('ProductDetailsScreen_NextButton');
    const productAnimation = screen.getByTestId('ProductDetailsScreen_Animation');

    expect(nextButton).toBeTruthy();
    expect(inputField).toBeTruthy();
    expect(productAnimation).toBeTruthy();

  });

});
