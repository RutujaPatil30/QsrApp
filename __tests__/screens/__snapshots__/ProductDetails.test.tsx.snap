// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking ProductDetailsScreen  should render the UI "ProductDetailsScreen"  1`] = `
<View
  style={
    {
      "backgroundColor": "#da291c",
      "flex": 1,
    }
  }
>
  <AnimatableView
    animation="fadeIn"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingTop": 120,
        }
      }
    >
      <RCTScrollView
        collapsable={false}
        handlerTag={2}
        handlerType="NativeViewGestureHandler"
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        style={
          {
            "backgroundColor": "#ffffff",
            "borderTopLeftRadius": 66.5,
            "borderTopRightRadius": 66.5,
            "padding": 33.5,
            "top": 320,
          }
        }
        waitFor={
          [
            {
              "current": null,
            },
          ]
        }
      >
        <View>
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 22,
                "fontWeight": "bold",
              }
            }
          >
            Big Mac
          </Text>
          <Text
            style={
              {
                "color": "#ffc72c",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Rs. 
            5.99
          </Text>
          <Text
            style={
              {
                "fontSize": 15,
                "paddingVertical": 20,
              }
            }
          >
            A big, tasty burger with two beef patties and special sauce.
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Tambahkan Catatan
            </Text>
            <Text
              style={
                {
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginLeft": 6.5,
                }
              }
            >
              (Optional)
            </Text>
          </View>
          <TextInput
            onChangeText={[Function]}
            placeholder="Cth. Tanpa keju dan tomat"
            placeholderTextColor="#808080"
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "#e2e2e2",
                "borderRadius": 20,
                "marginVertical": 33.5,
                "padding": 20,
                "width": 675,
              }
            }
            testID="ProductDetailsScreen_InputField"
            value=""
          />
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "padding": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#da291c",
            "borderRadius": 112.5,
            "height": 76,
            "justifyContent": "center",
            "opacity": 1,
            "width": 97.5,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#ffffff",
                "fontSize": 25,
              },
              undefined,
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          Û∞êï
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#da291c",
            "borderColor": "#da291c",
            "borderRadius": 53.5,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 562.5,
          }
        }
        testID="ProductDetailsScreen_NextButton"
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              {
                "color": "#ffffff",
              },
            ]
          }
        >
          Pesan Sekarang
        </Text>
      </View>
    </View>
    <AnimatableView
      animation="slideInUp"
      style={
        {
          "alignItems": "center",
          "left": 6,
          "position": "absolute",
          "right": 0,
        }
      }
      testID="ProductDetailsScreen_Animation"
    >
      <RCTImageView
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        internal_analyticTag={null}
        resizeMode="contain"
        source={
          [
            {
              "height": undefined,
              "uri": undefined,
              "width": undefined,
            },
          ]
        }
        style={
          {
            "height": 400,
            "overflow": "hidden",
            "resizeMode": "contain",
            "top": 106.5,
            "width": 533.5,
          }
        }
      />
    </AnimatableView>
  </AnimatableView>
</View>
`;
